internal/embed/defaults/obol-stack-defaults/templates/obol-agent.yaml

```diff
@@ -0,0 +1,283 @@
+    verbs: ["get", "list", "watch", "create", "update", "patch"]
+  - apiGroups: ["batch"]
+    resources: ["jobs", "cronjobs"]
+    verbs: ["get", "list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["pods/log"]
+    verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: obol-agent-binding
+  namespace: monitoring
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: obol-agent-role
+subjects:
+  - kind: ServiceAccount
+    name: obol-agent
+    namespace: agent
+
+---
+#------------------------------------------------------------------------------
+# Deployment - Obol Agent Application
+# The agent provides AI-powered Kubernetes and Obol cluster management via MCP
+#------------------------------------------------------------------------------
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: obol-agent
+  namespace: agent
+  labels:
+    app: obol-agent
+spec:
+  replicas: 1  # Single instance deployment
+  selector:
+    matchLabels:
+      app: obol-agent
+  template:
+    metadata:
+      labels:
+        app: obol-agent
+    spec:
+      serviceAccountName: obol-agent  # Uses the ServiceAccount created above for RBAC
+      containers:
+        - name: obol-agent
+          image: us-east4-docker.pkg.dev/prj-d-playgrounds-f0cb/obol-agent/obol-agent-ag-ui:latest
+          imagePullPolicy: Always  # Always pull latest image
+          ports:
+            - name: http
+              containerPort: 8000
+              protocol: TCP
+          env:
+            # REQUIRED: Google API key from Kubernetes secret
+            # Secret created via: obol stack up --google-api-key=<key>
+            - name: GOOGLE_API_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: obol-agent-google-api-key
+                  key: GOOGLE_API_KEY
+                  optional: true  # Allow deployment even if secret doesn't exist
+
+            # PUBLIC_MODE controls Kubernetes MCP access
+            # false = Enable Kubernetes API access (uses RBAC permissions above)
+            # true = Disable Kubernetes API access (for public deployments)
+            - name: PUBLIC_MODE
+              value: "false"
+
+          # Health checks ensure the pod is ready to receive traffic
+          livenessProbe:
+            httpGet:
+              path: /health
+              port: http
+            initialDelaySeconds: 30
+            periodSeconds: 10
+          readinessProbe:
+            httpGet:
+              path: /health
+              port: http
+            initialDelaySeconds: 10
+            periodSeconds: 5
+
+          # Resource limits prevent the agent from consuming too many cluster resources
+          resources:
+            requests:  # Minimum guaranteed resources
+              cpu: 500m      # 0.5 CPU cores
+              memory: 1Gi    # 1 GiB RAM
```

Comment:
i think in theory not specifying a min allows the stack to be as light as possible. Do we want to reserve a full gb of ram for an agent people might not use all the time? if it doesn't break stuff i'd remove